src/runtime_src/core/common/sensor.cpp

```diff
@@ -160,9 +160,16 @@ read_data_driven_electrical(const std::vector<xq::sdm_sensor_info::data_type>& c
     auto desc = tmp.label;
     pt.put("id", desc);
     pt.put("description", desc);
-    pt.put("voltage.volts", xrt_core::utils::format_base10_shiftdown3(tmp.input));
-    pt.put("voltage.max", xrt_core::utils::format_base10_shiftdown3(tmp.max));
-    pt.put("voltage.average", xrt_core::utils::format_base10_shiftdown3(tmp.average));
+    /*
+     * Use below calculation for sensor values:
+     * actual sensor value = sensor_value * (10 ^ (unit_modifier))
+     * Example: Sensor Value 12000, Units “Volts” & Unit Modifier -3 received.
+     * So, actual sensor value => 12000 * 10 ^ (-3) = 12 Volts.
+     */
+    auto unitm = pow(10, tmp.unitm);
+    pt.put("voltage.volts", static_cast<double>(tmp.input) * unitm);
```

Comment:
You can define a simple function which takes unit modifier , value & returns the final value. Use that function all the places so that, if we need to consider units in future, we can just update that function.